honeycomb-core/src/cmap2/embed.rs

```diff
@@ -18,24 +18,31 @@ impl<T: CoordsFloat> CMap2<T> {
         self.vertices.n_attributes()
     }
 
+    #[allow(clippy::missing_errors_doc)]
     /// Fetch vertex value associated to a given identifier.
     ///
     /// # Arguments
     ///
     /// - `vertex_id: VertexIdentifier` -- Identifier of the given vertex.
     ///
-    /// # Return
+    /// # Return / Errors
     ///
-    /// Return a reference to the [Vertex2] associated to the ID.
+    /// The method returns:
+    /// This method return a `Result` taking the following values:
+    /// - `Ok(v: Vertex2)` if there is a vertex associated to this ID.
+    /// - `Err(CMapError::UndefinedVertexID)` -- otherwise
     ///
     /// # Panics
     ///
-    /// The method may panic if no vertex is associated to the specified index, or the ID lands
-    /// out of bounds.
+    /// The method may panic if:
+    /// - the index lands out of bounds
+    /// - the index cannot be converted to `usize`
```

Comment:
So Panics + Result?
I do not understand why panic cases are not handled by Result.

Reply:
Errors and panics cover different things here:
- The error can happen in a "legitimate" case, where the user is manually building the topology using `i_link` methods. Because the user can retrieve a valid vertex ID (pointing to nothing), I think he should be able to handle the issue without crashing the program.
- Panics come directly from the `AttrSparseVec::get` implementation. Because the storage is wired to the map correctly, the only way to land out of bounds / fail the cast is to use arbitrary ID values.